DigitalOffice/Services/CheckRightsService/Broker/Consumers/CheckIfUserHasRightConsumer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentValidation;
+using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.CheckRightsService.Repositories.Interfaces;
+using LT.DigitalOffice.Kernel.Broker;
+using MassTransit;
+
+namespace LT.DigitalOffice.CheckRightsService.Broker.Consumers
+{
+    public class CheckIfUserHasRightConsumer : IConsumer<ICheckIfUserHasRightRequest>
+    {
+        private readonly IValidator<ICheckIfUserHasRightRequest> validator;
+        private readonly ICheckRightsRepository repository;
+
+        public CheckIfUserHasRightConsumer(IValidator<ICheckIfUserHasRightRequest> validator, ICheckRightsRepository repository)
+        {
+            this.validator = validator;
+            this.repository = repository;
+        }
+
+        public async Task Consume(ConsumeContext<ICheckIfUserHasRightRequest> context)
+        {
+            try
+            {
+                validator.ValidateAndThrow(context.Message);
+                await context.RespondAsync<IOperationResult<bool>>(new
+                {
+                    Body = repository.CheckIfUserHasRight(context.Message),
+                    IsSuccess = true,
+                    Errors = new List<string>()
+                });
+            }
+            catch (Exception exception)
+            {
+                await context.RespondAsync<IOperationResult<bool>>(new
+                {
+                    Body = false,
+                    IsSuccess = false,
+                    Errors = new List<string> {exception.Message}
+                });
```

Comment:
DRY, move await context.RespondAsync<IOperationResult<bool>> behind construction try{}catch{}.

Reply:
I don't really like this idea, because I think that anonymous types should be used only in the scope where they were initialized. Your option involves creating a variable of type object outside try, catch

Reply:
Added wrapper for OperationResult

Reply:
And again I agree with @djino97. Move context.RespondAsync<IOperationResult> behind construction try{}catch{} and use OperationResultWrapper. Remove validator from here at all

Reply:
fixed